packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -3920,8 +3920,25 @@ export class ITxnSubmit extends Op {
       );
     }
 
+    // initial contract account.
+    const appID = this.interpreter.runtime.ctx.tx.apid;
+    const contractAddress = getApplicationAddress(appID as number);
+    const contractAccount = {
+      addr: contractAddress,
+      sk: Buffer.from([])
+    };
+
     // get execution txn params (parsed from encoded sdk txn obj)
-    const execParams = parseEncodedTxnToExecParams(this.interpreter.subTxn, this.interpreter, this.line);
+    // singer will be contractAccount
+    const execParams = encTxToExecParams(
+      this.interpreter.subTxn,
+      {
+        sign: types.SignType.SecretKey,
+        fromAccount: contractAccount
+      },
```

Comment:
Why not set this inside in `encTxToExecParams`?

Reply:
> Why not set this inside in `encTxToExecParams`?

`encTxToExecParams` more general now. We use it to convert `TransactionAndSign` to `ExecParams`, so `sign` can be Lsig. I think keep it in parameters of method is better.